app/lib/widgets/wallets/add_wallet.dart

```diff
@@ -62,13 +101,20 @@ class _NewWalletState extends State<NewWallet> {
   }
 
   bool _validateSecret(String walletSecret) {
+    List<String> hexSeeds = getWalletHexSeeds();
+    String? walletSeed = generateHexSeed(walletSecret);
```

Comment:
this should happen after validating the input first

Reply:
but if the input is valid the wallet will be added without checking for duplicates

Reply:
I think the right flow is to make duplicated condition second one.
If the secret is a duplicate, there's no point in further validating.